x/wasm/keeper/api.go

```diff
@@ -22,25 +26,39 @@ const (
 var (
 	costHumanize            = DefaultGasCostHumanAddress * types.DefaultGasMultiplier
 	costCanonical           = DefaultGasCostCanonicalAddress * types.DefaultGasMultiplier
+	costValidate            = DefaultGasCostCanonicalAddress * types.DefaultGasMultiplier
 	costJSONDeserialization = wasmvmtypes.UFraction{
 		Numerator:   DefaultDeserializationCostPerByte * types.DefaultGasMultiplier,
 		Denominator: 1,
 	}
 )
 
-func humanAddress(canon []byte) (string, uint64, error) {
+func humanizeAddress(canon []byte) (string, uint64, error) {
 	if err := sdk.VerifyAddressFormat(canon); err != nil {
 		return "", costHumanize, err
 	}
 	return sdk.AccAddress(canon).String(), costHumanize, nil
 }
 
-func canonicalAddress(human string) ([]byte, uint64, error) {
+func canonicalizeAddress(human string) ([]byte, uint64, error) {
 	bz, err := sdk.AccAddressFromBech32(human)
 	return bz, costCanonical, err
 }
 
+func validateAddress(human string) (uint64, error) {
+	canonicalized, err := sdk.AccAddressFromBech32(human)
+	if err != nil {
+		return costValidate, err
+	}
+	// AccAddressFromBech32 already calls VerifyAddressFormat, so we can just humanize and compare
+	if canonicalized.String() != human {
+		return costValidate, errors.New("address not normalized")
+	}
+	return costValidate, err
```

Comment:
The end of the function is supposed to be the success case, right?

```suggestion
	return costValidate, nil
```